src/server/server/dal.py

```diff
@@ -36,44 +53,73 @@ def get_features(feature_set_id):
     return [feature.name for feature in db_feature_set.features]
 
 
-def _to_feature_vector(db_feature_vector):
-    if db_feature_vector is None:
-        return None
+def save_user_feature_vector(user, feature_vector):
+    db_feature_vector = _to_db_feature_vector(feature_vector)
+    db_feature_vector.key = user.feature_vector_db_key
+    db_feature_vector.put()
+
+
+def get_user_feature_vector(user):
+    return _to_feature_vector(user.feature_vector_db_key.get())
 
+
+def _to_feature_vector(db_feature_vector):
     vector = db_feature_vector.vector
     feature_set_db_key = db_feature_vector.feature_set_key
-    labels = [feature.name for feature in feature_set_db_key.get().features]
-    return FeatureVector(vector=vector, labels=labels, feature_set_id=feature_set_db_key.id())
+    feature_set = feature_set_db_key.get()
+    labels = [feature.name for feature in feature_set.features]
+    return struct.FeatureVector(vector=vector, labels=labels, feature_set_id=feature_set_db_key.id())
```

Comment:
To be consistent with the way other "front-end structs" are instantiated, instead of directly calling the constructor, call a static "factory" method ?